readme.md

```diff
@@ -24,110 +33,269 @@ Or you can run it from Maven directly using the Spring Boot Maven plugin. If you
 
-## Looking for something in particular?
 
-|Spring Boot Configuration | Class or Java property files  |
-|--------------------------|---|
-|The Main Class | [PetClinicApplication](https://github.com/spring-projects/spring-petclinic/blob/main/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java) |
-|Properties Files | [application.properties](https://github.com/spring-projects/spring-petclinic/blob/main/src/main/resources) |
-|Caching | [CacheConfiguration](https://github.com/spring-projects/spring-petclinic/blob/main/src/main/java/org/springframework/samples/petclinic/system/CacheConfiguration.java) |
+```java
+// Binding tarantool function
+@Query(function = "find_owner_by_id")
+Owner findOwnerById(UUID id);
 
-## Interesting Spring Petclinic branches and forks
+// Using default CrudRepository operations
+Owner save(Owner owner);
+```
 
-The Spring Petclinic "main" branch in the [spring-projects](https://github.com/spring-projects/spring-petclinic)
-GitHub org is the "canonical" implementation, currently based on Spring Boot and Thymeleaf. There are
-[quite a few forks](https://spring-petclinic.github.io/docs/forks.html) in a special GitHub org
-[spring-petclinic](https://github.com/spring-petclinic). If you have a special interest in a different technology stack
-that could be used to implement the Pet Clinic then please join the community there.
+Data from tarantool is mapped from the space description in the form space_object:format to the java description annotated with @Tuple. And the fields themselves are annotated with @Field.
 
+```java
+@Tuple("visits")
+public class Visit extends BaseEntity {
 
-## Interaction with other open source projects
+    @Field(name = "visit_date")
+    @DateTimeFormat(pattern = "yyyy-MM-dd")
+    private LocalDate date;
 
-One of the best parts about working on the Spring Petclinic application is that we have the opportunity to work in direct contact with many Open Source projects. We found some bugs/suggested improvements on various topics such as Spring, Spring Data, Bean Validation and even Eclipse! In many cases, they've been fixed/implemented in just a few days.
-Here is a list of them:
+    @NotEmpty
+    @Field(name = "description")
+    private String description;
+...
+```
 
-| Name | Issue |
-|------|-------|
-| Spring JDBC: simplify usage of NamedParameterJdbcTemplate | [SPR-10256](https://jira.springsource.org/browse/SPR-10256) and [SPR-10257](https://jira.springsource.org/browse/SPR-10257) |
-| Bean Validation / Hibernate Validator: simplify Maven dependencies and backward compatibility |[HV-790](https://hibernate.atlassian.net/browse/HV-790) and [HV-792](https://hibernate.atlassian.net/browse/HV-792) |
-| Spring Data: provide more flexibility when working with JPQL queries | [DATAJPA-292](https://jira.springsource.org/browse/DATAJPA-292) |
+Data is returned from the router instance. Using custom functions or crud. Configuration how to connect to tarantool router:
```

Comment:
Style fix:

Data is obtained from cluster via custom functions lua functions or 'crud' library that execute on 'router' instance. Here is an example of configuration to establish connection to tarantool router instance: